src/CSI.IBTA.BenefitsService/Services/DecodingService.cs

```diff
@@ -0,0 +1,54 @@
+﻿using CSI.IBTA.UserService.Interfaces;
+using CSI.IBTA.Shared.DTOs;
+using System.Security.Cryptography;
+using System.Text.Json;
+using System.Text;
+
+namespace CSI.IBTA.UserService.Services
+{
+    internal class DecodingService : IDecodingService
+    {
+        private readonly ILogger<DecodingService> _logger;
+        private readonly byte[] _aesKey;
+
+        public DecodingService(IConfiguration configuration, ILogger<DecodingService> logger)
+        {
+            _logger = logger;
+            var aesKey = configuration.GetSection("AesKey").Get<byte[]>();
+            if (aesKey == null || aesKey.Length == 0)
+            {
+                _logger.LogError("AesKey is missing in appsettings.json");
+                throw new InvalidOperationException("AesKey is missing in appsettings.json");
+            }
+            _aesKey = aesKey;
+        }
+
+        public GenericResponse<EmployerEmployeeDto> GetDecodedEmployerEmployee(byte[] encryptedData)
```

Comment:
Why do we need decoding/encoding here, what kind of requirement is implemented here ?

Reply:
Benefits service does not have access to Users database. We need to check if employee resource is owned by employer.
Siarhei suggested to encrypt data object and pass it to the request body to solve this issue.